pydantic/error_wrappers.py

```diff
@@ -51,8 +51,9 @@ def dict(self, *, loc_prefix: Optional[Tuple[str, ...]] = None) -> Dict[str, Any
 class ValidationError(ValueError):
     __slots__ = ('raw_errors',)
 
-    def __init__(self, errors: Sequence[ErrorList]) -> None:
+    def __init__(self, errors: Sequence[ErrorList], obj_or_type: Any) -> None:
```

Comment:
```suggestion
    def __init__(self, errors: Sequence[ErrorList], model: Type[Any]) -> None:
```

Then you'll need to call `type(...)` in a couple of places, but I think this makes more sense.